### DISCUSSION TITLE
picker的事件类型定义与文档描述的不符

### DISCUSSION THREAD

**Discussion by zzxqd** [youzan/vant#11740] [General]
### 重现链接

https://codesandbox.io/s/vant-3-issue-template-forked-dmu81k?file=/src/App.vue

### Vant 版本

4.1.2

### 描述一下你遇到的问题。

picker的onConfim等事件的类型定义
![截图 2023-04-07 15 09 45](https://user-images.githubusercontent.com/23022401/230560280-faf327ad-2ec8-4055-b985-839044edfd76.png)
文档描述中的类型定义
![截图 2023-04-07 15 10 44](https://user-images.githubusercontent.com/23022401/230574495-03355e5e-42df-4c75-84e1-88ca2407389d.png)

### 重现步骤

import {Picker} from 'vant'
点击进入Picker的类型定义文件中就能看到

**Comment by chenjiahan** [comment]
目前 Vant 里的事件类型都使用了 Vue 生成的默认类型，因为虽然 Vue 提供了 defineEmits 来声明事件类型，但是只能在 `script setup` 中使用。

在 TSX 组件里，似乎还没有很合适的方式来定义 emits 的类型，这个 issue 先转为 Discussion 吧，后面有合适的事件类型声明方式后，我们会进行优化。
